app/src/main/java/com/example/leagueoflegends/misc/MainAdapter.kt

```diff
@@ -0,0 +1,52 @@
+package com.example.leagueoflegends.misc
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.example.leagueoflegends.R
+import com.example.leagueoflegends.model.Response
+import com.squareup.picasso.Picasso
+import kotlinx.android.synthetic.main.champion_main_item.view.*
+
+class MainAdapter: RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+
+
+    private var champions  = emptyArray<Response.Champion>()
```

Comment:
This response is a model layer class and it shouldn't be used in the view. If the API changes the response / you wanna switch it to a DB,  you'll have to touch too many layers of your code. I suggest mapping them when it arrives in the presenter, preferably in a separate mapping class.